src/components/Dropdown/Dropdown.tsx

```diff
@@ -414,30 +484,38 @@ export default class Dropdown extends AutoControlledComponent<
           { ...this.props, searchQuery: changes.inputValue },
         )
         return changes
+      case Downshift.stateChangeTypes.blurButton:
+        // Downshift closes the list by default on trigger blur. It does not support the case when dropdown is
+        // single selection and focuses list on trigger click/up/down/space/enter. Treating that here.
+        if (state.isOpen && document.activeElement === this.listNode) {
+          return {} // won't change state in this case.
+        }
       default:
         return changes
     }
   }
 
-  private getItemsFilteredBySearchQuery = (items: ShorthandValue[]): ShorthandValue[] => {
-    const { itemToString, multiple, search } = this.props
+  private getItemsFilteredBySearchQuery = (): ShorthandValue[] => {
+    const { items, itemToString, multiple, search } = this.props
     const { searchQuery, value } = this.state
+    let filteredItems = items
 
-    const nonSelectedItems = items.filter(item =>
-      multiple ? (value as ShorthandValue[]).indexOf(item) === -1 : true,
-    )
-
-    const itemsMatchSearchQuery = nonSelectedItems.filter(item =>
-      search
-        ? _.isFunction(search)
-          ? search(itemsMatchSearchQuery, searchQuery)
-          : itemToString(item)
-              .toLowerCase()
-              .indexOf(searchQuery.toLowerCase()) !== -1
-        : true,
-    )
+    if (multiple) {
+      filteredItems = filteredItems.filter(item => (value as ShorthandValue[]).indexOf(item) === -1)
```

Comment:
```suggestion
      filteredItems = _.pull(filteredItems, value)
```

Reply:
doing that will mutate `filteredItems`, and also `items`, which is not what I want to do. to use `pullAll` I need to first create a deep copy of the array and use `pullAll` on that copy. unless you think of something better :)

Reply:
You can use `_.difference()` in this case